confirmation_rule.py

```diff
@@ -125,7 +125,10 @@ def find_latest_confirmed_descendant(store, latest_confirmed_root) -> Root:
         )
         
         if (block_epoch > compute_epoch_at_slot(latest_confirmed_slot)
-            and not will_checkpoint_be_justified(store, checkpoint)):
+            and 
+                (not will_checkpoint_be_justified(store, checkpoint) or
+                store.unrealized_justifications[get_head(store)].epoch + 1 >= current_epoch)
```

Comment:
```suggestion
                store.unrealized_justifications[get_head(store)].epoch + 1 < current_epoch)
```

With the above fix, it stops confirming if either checkpoint won’t be justified or justification in the view of the head is too old. I think this was the intention

Reply:
I also suggest we move this check to the outer function, because as long as it is `False` safety of any confirmed block is questionable regardless of its epoch.

Reply:
With `H = get_head()` consider the case when `UJ(H).epoch >= current_epoch - 1`, but `H` isn’t lmd-confirmed yet, in the next slot `get_head() = H'` and `UJ(H').epoch >= current_epoch - 2`. Assuming synchrony and `beta < 1/3`, this can’t happen, right? If it can then we will have to run this check over the most recent lmd-confirmed descendant of the block in question

Reply:
Actually, from the paper:
> Assumption 5.3 essentially says that Byzantine validators cannot prevent an FFG vote sent by a validator that is not caught committing a slashable oﬀence from being included in a canonical block for an entire epoch.

As long as there are enough FFG votes we assume they will be included into canonical chain block, given that assumption it doesn’t matter if the `head` with which we did the check would have been re-orged after that